climada/hazard/tc_tracks_forecast.py

```diff
@@ -74,6 +75,9 @@
 
 LOGGER = logging.getLogger(__name__)
 
+MISSING_DOUBLE = ec.CODES_MISSING_DOUBLE
+MISSING_LONG = ec.CODES_MISSING_LONG
+"""Missing double and integers in ecCodes """
```

Comment:
Is this a module constant or something that should be adjusted as configurables?

Reply:
interesting question. we'd only want to change these missing values if ECMWF change their missing values in the future (very unlikely imo). in that situation the current code would allow the user to update ecCodes manually without having to modify any code themselves.

on the other hand, with the CLIMADA release cycle it might be three months before it's fixed in CLIMADA, and the user may want to manually specify these values instead. but this situation is really unlikely, so i wouldn't prepare for it.

Reply:
Ok, I see. But how does this work with backward compatibility? Will this code just stop working after every update of CLIMADA or ECMWF?

Reply:
it'll stop working after every big update to ECMWF BUFR formats, but there's no way to prepare for that (e.g. the latest update replaced ~12 variables with a single integer that gives the same variables via a lookup).

the hope is that, by using ECMWF's ecCodes to read files, some breaking updates to the forecast messages can be handled by updating ecCodes instead of rewriting CLIMADA.

backward compatibility will always be a nightmare. we have two options, i guess:
- people have to check out old versions of CLIMADA to read old-format BUFR messages (this is how it works now)
- we add a 'version' parameter to `read_one_bufr_tc`, and point to different methods based on that parameter

i'd suggest sticking with what we've got until someone needs to work with archived messages, since that's not yet a use case

Reply:
So this the the missing doubles and missing integers defined in the ECMWF's bufr files. Theses missing constants are defined in the ecCodes constant values. I assume that if they change the bufr formats, they will also update the constants value from their library.

Reply:
What do you mean with "i'd suggest sticking with what we've got until someone needs to work with archived messages, since that's not yet a use case"? 

I would say that having access to past/archived forecasts is absolutely crucial for research. Or I am missing something here?

Reply:
Using ecCodes can access to past forecast tracks. Their recent changes of the BUFR files involve the change of the 'descriptors' (see [here](https://confluence.ecmwf.int/display/FCST/Update+to+Tropical+Cyclone+tracks)), that is a 6 digits number that is used to access the information. The old codes were written evolving the old descriptors and there's is no easy fix to read the new forecast tracks.

The missing constants defined are the same for past and present BUFR files. In a very unlikely case that they may change the values of these missing constants, but then it should be their responsibility to update `eccodes.CODES_MISSING_DOUBLE` and `eccodes.CODES_MISSING_LONG`.

Reply:
we also haven't seen a public archive of forecasts, so if they exist i assume they're behind a paywall